client/transport/src/ws/mod.rs

```diff
@@ -206,11 +217,19 @@ impl TransportReceiverT for Receiver {
 	type Error = WsError;
 
 	/// Returns a `Future` resolving when the server sent us something back.
-	async fn receive(&mut self) -> Result<String, Self::Error> {
+	async fn receive(&mut self) -> Result<ReceivedMessage, Self::Error> {
 		let mut message = Vec::new();
-		self.inner.receive_data(&mut message).await?;
-		let s = String::from_utf8(message).expect("Found invalid UTF-8");
-		Ok(s)
+
+		loop {
+			let recv = self.inner.receive(&mut message).await?;
+
+			if let Incoming::Data(_) = recv {
+				let s = String::from_utf8(message).expect("Found invalid UTF-8");
```

Comment:
I think this expect is okay because we send only messages as `text` but it's better that we just return an error here instead as more transports than `WebSocket` has been added.

This is probably my own fault but would be great if that is fixed too.